tests/kernel/threads/lifecycle/thread_init/src/test_thread_init.c

```diff
@@ -194,3 +197,17 @@ void test_kinit_coop_thread(void)
 	/*wait for thread to exit*/
 	k_sem_take(&end_sema, K_FOREVER);
 }
+
+/*test case main entry*/
+void test_main(void)
+{
+	k_thread_access_grant(k_current_get(), &thread_preempt, &stack_preempt,
+			      &start_sema, &end_sema, NULL);
+
+	ztest_test_suite(test_thread_init,
+			 ztest_user_unit_test(test_kdefine_preempt_thread),
+			 ztest_user_unit_test(test_kdefine_coop_thread),
+			 ztest_user_unit_test(test_kinit_preempt_thread),
+			 ztest_unit_test(test_kinit_coop_thread));
```

Comment:
same doubt again, why coop threads can't run in user mode?
also why coop thread test at 209 is run in user mode?

Reply:
this test was skipped, user threads can't create new threads that are of greater priority than they are

the other coop test works fine